service/src/test/java/uk/nhs/adaptors/gp2gp/utils/DateFormatUtilTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.nhs.adaptors.gp2gp.utils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil.toHl7Format;
+
+import java.util.stream.Stream;
+
+import org.hl7.fhir.dstu3.model.Observation;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+
+@ExtendWith(MockitoExtension.class)
+public class DateFormatUtilTest {
+    private static final FhirParseService FHIR_PARSER = new FhirParseService();
+    private static final String INSTANT_OBSERVATION_TEMPLATE = "{\"resourceType\": \"Observation\", \"issued\": \"%s\"}";
+    private static final String DATE_OBSERVATION_TEMPLATE = "{\"resourceType\": \"Observation\", \"valueDate\": \"%s\"}";
+    private static final String DATETIME_OBSERVATION_TEMPLATE = "{\"resourceType\": \"Observation\", \"valueDateTime\": \"%s\"}";
+
+    @ParameterizedTest
+    @MethodSource("instantParams")
+    public void When_FormattingInstantToHl7_Expect_Hl7InUkZone(String input, String expected) {
+        String observationJson = String.format(INSTANT_OBSERVATION_TEMPLATE, input);
+        Observation observation = FHIR_PARSER.parseResource(observationJson, Observation.class);
+
+        String actual = toHl7Format(observation.getIssuedElement());
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @ParameterizedTest
+    @MethodSource("dateParams")
+    public void When_FormattingDateTypeToHl7_Expect_Hl7InUkZone(String input, String expected) {
```

Comment:
This is testing InstantType not DateType. It appears that "valueDate" (DateType) is not valid in Observation (or any resource used by GP Connect) so this test can be removed. Don't forgot to remove the unused template.

Reply:
Use a different resource type that does support dateType